### PR TITLE
[client][python] getLastMessageIdAsync C binding

### PR DESCRIPTION
**Pull request by komalatammal**
### Motivation

Python function getLastMessageId

It is a C binding for https://github.com/apache/pulsar/pull/16182 to implement get_last_message_id() in Python client.

### Modifications

Add Python/C binding code for get_last_message_id()

### Verifying this change

It compiles.
- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (yes)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 

  
- [x] `doc` 
Python Doc is updated in __init__.py

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by BewareMyPower — APPROVED** [review]

File: pulsar-client-cpp/README.md [hunk 1]
```diff
@@ -280,7 +280,7 @@ ${PULSAR_PATH}/pulsar-test-service-stop.sh
 
 ## Requirements for Contributors
 
-It's recommended to install [LLVM](https://llvm.org/builds/) for `clang-tidy` and `clang-format`. Pulsar C++ client use `clang-format` 6.0+ to format files. 
+It's required to install [LLVM](https://llvm.org/builds/) for `clang-tidy` and `clang-format`. Pulsar C++ client use `clang-format` 6.0+ to format files.  `make format` will automatically format the files.
```
[the remark this answers is not in the document]
Comment (zzzming) [reply pulsar-client-cpp/README.md]: updated

**Review by eolivelli — APPROVED** [review]

**Review by michaeljmarshall — APPROVED** [review]
LGTM

**Comment by michaeljmarshall** [comment]
@komalatammal @eolivelli @BewareMyPower - it'd be nice to cherry pick this to 2.11 since it is a feature that is already in the Java client api. Do you think it is too late to get this change into that branch?

**Comment by BewareMyPower** [comment]
@michaeljmarshall I think it needs a discussion. There was a similar discussion here: https://github.com/apache/pulsar/pull/15822#issuecomment-1206270637 /cc @nodece 

The core question is that should we cherry-pick all these new features of C++/Python clients to release branches?

My personal answer is no. The feature catch up should not be cherry-picked. For the same reason, if users want some new features of a specific component, they have to upgrade the major version.

Regarding 2.11.0 release, I think you can reply in the mail list. I see branch-2.11 has already been created.

**Comment by michaeljmarshall** [comment]
> My personal answer is no. The feature catch up should not be cherry-picked. For the same reason, if users want some new features of a specific component, they have to upgrade the major version.

I agree with this line of thinking. My only reason for asking in this case is because branch-2.11 was only just recently created and the RC has not yet been tagged. I am not going to ask on the ML because I'll just let this target 2.12.

**Comment by nodece** [comment]
@BewareMyPower @michaeljmarshall Thanks for your explanation!  I have a question about the broker and client of old version, if the broker supports the xx feature, but the client doesn't support it, shouldn't we cherry-pick it?

**Comment by BewareMyPower** [comment]
>  if the broker supports the xx feature, but the client doesn't support it

IMO no as I've said:

> The feature catch up should not be cherry-picked.

**Comment by michaeljmarshall** [comment]
> @BewareMyPower @michaeljmarshall Thanks for your explanation! I have a question about the broker and client of old version, if the broker supports the xx feature, but the client doesn't support it, shouldn't we cherry-pick it?

One reason I think it is unnecessary to backport features to older versions of the client is that the protocol has a feature handshake built in that is supposed to enable different versions of the clients and servers to work together seamlessly. Therefore, if a user wants to get a new feature in the client that was already supported on the broker in an older version, they should be able to upgrade without issue. That being said, I am not sure that we document the client/server compatibility anywhere and I am not sure how good the test coverage is for the many permutations of client versions.

**Comment by nodece** [comment]
Ok, I see.

**Pull request merged**